lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -3652,8 +3652,11 @@ ResolvedType GenericSignatureBuilder::maybeResolveEquivalenceClass(
                                    wantExactPotentialArchetype);
     if (!resolvedBase) return resolvedBase;
     // If the base is concrete, so is this member.
-    if (resolvedBase.getAsConcreteType())
+    if (resolvedBase.getAsConcreteType()) {
+      if (type->isTypeParameter())
```

Comment:
This still seems like a bit of a hack to me. Generic parameter types are handled above but, it feels like this type parameter type should never make it here (I have no basis for thinking that, though). Could someone with more knowledge of this bit of code chime in, maybe? 

cc @jckarter based on git blame

Reply:
I don't recall, sorry. If you can remove it and it still works, then by all means remove it.

Reply:
No worries. It's not removing anything as much as conditionally returning an unresolved nullptr. It seems to work, though, so it's probably OK.